### DISCUSSION TITLE
LDAP Auth user

### DISCUSSION THREAD

**Discussion by DobroFenix** [nilsteampassnet/TeamPass#4514] [General]
Hi!
I saw that in recent changes you have changed the interaction with LDAP.
Help please configure LDAP domain user authentication for Active Directory (Windows).
Everything looks set up correctly, the check is successful, but from the initial form of authentication it is not possible to log in under the user's domain account.
LDAP config:
![image](https://github.com/user-attachments/assets/a3677a9c-4ba9-4609-8fef-e3cd8160182e)
![image](https://github.com/user-attachments/assets/38fa70ec-36f1-410a-9d83-374da7fe3b2b)

After trying to authenticate under the domain user, I get the error: Login credentials do not correspond!
At the same time, the connection check at the bottom of the LDAP connection configuration form successfully checks the authenticators of the same user.
![image](https://github.com/user-attachments/assets/480f26e3-cf4c-4ffe-af88-62d99a264901)

**Comment by DobroFenix** [comment]
I change in /var/www/teampass/includes/config/include.php:
define('DEBUG', true);
define('DEBUGLDAP', true); //Can be used in order to debug LDAP authentication
Where do I look for these LDAP user validation logs when I log on?

**Reply by WoutAndAbout** [reply]
Did you find out?

**Comment by DobroFenix** [comment]
![image](https://github.com/user-attachments/assets/a9d5f1cc-f61f-4404-92d9-c6741874d96d)
Still does not let the domain user into the system.

**Comment by DobroFenix** [comment]
How do I transfer all users from AD?
No temporary security code.

**Comment by vura** [comment]
Hi @DobroFenix 

My configuration.

![ldapteampass](https://github.com/user-attachments/assets/40964126-e0a0-4a61-b41e-a945eb6201f5)


In the field: User Object Filter, I added the following filter: (memberOf=CN=G_TEAMPASS_INFRA,OU=Groups,OU=Organizational Structure - Groups,DC=xxx,DC=xx,DC=xxx,DC=xx)

Where G_TEAMPASS_INFRA is the group in my AD of users who will have access to teampass.

Would it be something like that?

**Reply by DobroFenix** [reply]
Thank you.
I think I have already passed this stage.
How now to make sure that all users from AD are transferred to TeamPass?
I can't imagine how to manually transfer many thousands of users.

LDAP synchronization works with me and shows a list of all users who should have access to TeamPass.
![image](https://github.com/user-attachments/assets/5f07a208-92f2-4360-b623-a701987641b4)

**Reply by DobroFenix** [reply]
My config:
![image](https://github.com/user-attachments/assets/3f5606b1-9828-4d81-976e-9d810de93df4)
Did the same just in case.

**Comment by vura** [comment]
Once you sync with AD and the users appear as you did, you need to add the users to tempass by pressing the add button.


![392819996-5f07a208-92f2-4360-b623-a701987641b4](https://github.com/user-attachments/assets/7399e268-e388-46a3-afc9-8321e1ec06ad)

**Reply by DobroFenix** [reply]
And this cannot be done automatically so that users themselves are added to TeamPass?
Or maybe there is some kind of registration so that the user can add himself to TeamPass?
This is problematic to do manually for several thousand people, just considering that someone will have to constantly sit and manually transfer new users from AD.

**Comment by DobroFenix** [comment]
I solved the problem by writing a separate converter that automatically transfers all users from AD to TeamPass.

**Reply by vura** [reply]
> Resolvi o problema escrevendo um conversor separado que transfere automaticamente todos os usuários do AD para o TeamPass.

Showw!!

**Reply by DobroFenix** [reply]
> > Resolvi o problema escrevendo um conversor separado que transfere automaticamente todos os usuários do AD para o TeamPass.
> 
> Showw!!

easy
json -> rename to php
[ad_to_teampass_for_vura.json](https://github.com/user-attachments/files/18230052/ad_to_teampass_for_vura.json)
